x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -259,19 +259,21 @@ private FunctionDefinition[][] functions() {
             // grouping functions
             new FunctionDefinition[] { def(Bucket.class, Bucket::new, "bucket", "bin"), },
             // aggregate functions
+            // since they declare two public constructors - one with filter (for nested where) and one without
+            // use casting to disambiguate between the two
             new FunctionDefinition[] {
-                def(Avg.class, Avg::new, "avg"),
-                def(Count.class, Count::new, "count"),
-                def(CountDistinct.class, CountDistinct::new, "count_distinct"),
-                def(Max.class, Max::new, "max"),
-                def(Median.class, Median::new, "median"),
-                def(MedianAbsoluteDeviation.class, MedianAbsoluteDeviation::new, "median_absolute_deviation"),
-                def(Min.class, Min::new, "min"),
-                def(Percentile.class, Percentile::new, "percentile"),
-                def(Sum.class, Sum::new, "sum"),
-                def(Top.class, Top::new, "top"),
-                def(Values.class, Values::new, "values"),
-                def(WeightedAvg.class, WeightedAvg::new, "weighted_avg") },
+                def(Avg.class, uni(Avg::new), "avg"),
```

Comment:
This was needed to help the compiler disambiguate which constructor to pick as there are multiple variants.